Org unit assignment does not work on API 34

See https://github.com/EyeSeeTea/user-extended-app/blob/master/src/components/batch-models-multi-select/BatchModelsMultiSelect.model.js#L44-L63

```
save(parents, allChildren, selectedIds, updateStrategy) {
        const api = this.d2.Api.getApi();
        const selectedChildren = _(allChildren)
            .keyBy("id")
            .at(...selectedIds)
            .compact()
            .value();
        const childrenForParents = this.getNewChildren(parents, selectedChildren, updateStrategy);
        const payload = this.getPayload(allChildren, _.zip(parents, childrenForParents));
        const metadataUrl = "metadata?importStrategy=UPDATE&mergeMode=REPLACE";

        return api.post(metadataUrl, payload).then(response => {
            if (response.status !== "OK") {
                console.error("Response error", response);
                throw new Error(response.status);
            } else {
                return response;
            }
        });
    }
```

API Error:

```
errorCode: "E4000" 
errorKlass: "org.hisp.dhis.user.UserCredentials" 
errorProperties: ["userCredentials"]
errorProperty: "userCredentials" 
mainKlass: "org.hisp.dhis.user.User" 
message: "Missing required property `userCredentials`." 
```

Console error:
```
BatchModelsMultiSelect.model.js:57 Response error {importParams: {…}, status: "ERROR", stats: {…}, typeReports: Array(1)}
(anonymous) @ BatchModelsMultiSelect.model.js:55
Promise.then (async)
value @ BatchModelsMultiSelect.model.js:53
value @ OrgUnitDialog.component.js:84
r.handleClick @ EnhancedButton.js:143
r @ ReactErrorUtils.js:25
c @ EventPluginUtils.js:80
f @ EventPluginUtils.js:95
g @ EventPluginHub.js:41
x @ EventPluginHub.js:53
n @ forEachAccumulated.js:22
processEventQueue @ EventPluginHub.js:252
i @ ReactEventEmitterMixin.js:15
handleTopLevel @ ReactEventEmitterMixin.js:27
h @ ReactEventListener.js:68
perform @ Transaction.js:132
batchedUpdates @ ReactDefaultBatchingStrategy.js:54
h @ ReactUpdates.js:95
dispatchEvent @ ReactEventListener.js:144
List.component.js:308 Error: ERROR
    at BatchModelsMultiSelect.model.js:58
_orgUnitAssignmentError @ List.component.js:307
(anonymous) @ OrgUnitDialog.component.js:93
Promise.catch (async)
value @ OrgUnitDialog.component.js:90
r.handleClick @ EnhancedButton.js:143
r @ ReactErrorUtils.js:25
c @ EventPluginUtils.js:80
f @ EventPluginUtils.js:95
g @ EventPluginHub.js:41
x @ EventPluginHub.js:53
n @ forEachAccumulated.js:22
processEventQueue @ EventPluginHub.js:252
i @ ReactEventEmitterMixin.js:15
handleTopLevel @ ReactEventEmitterMixin.js:27
h @ ReactEventListener.js:68
perform @ Transaction.js:132
batchedUpdates @ ReactDefaultBatchingStrategy.js:54
h @ ReactUpdates.js:95
dispatchEvent @ ReactEventListener.js:144
Api.js:244 Fetch finished loading: GET "http://localhost:8080/api/users?fields=displayName%7Crename(name)%2CshortName%2CfirstName%2Csurname%2Ccreated%2Cemail%2Cid%2CuserCredentials%5Busername%2Cdisabled%2CuserRoles%5Bid%2CdisplayName%5D%2ClastLogin%5D%2ClastUpdated%2Ccreated%2CdisplayDescription%2Ccode%2CpublicAccess%2Caccess%2Chref%2Clevel%2CuserGroups%5Bid%2CdisplayName%2CpublicAccess%5D%2CorganisationUnits%5Bid%2Ccode%2CdisplayName%5D%2CdataViewOrganisationUnits%5Bid%2Ccode%2CdisplayName%5D&paging=true&filter=name%3Ane%3Adefault&modelType=user&order=name%3Aiasc&query=tobe&page=1&pageSize=50".
```

--- a/src/List/organisation-unit-dialog/OrgUnitDialog.component.js
+++ b/src/List/organisation-unit-dialog/OrgUnitDialog.component.js
@@ -20,8 +20,12 @@ class OrgUnitDialog extends React.Component {
             getPayload: (allOrgUnits, pairs) => ({
                 users: pairs.map(([user, orgUnitsForUser]) =>
                     _m(getOwnedPropertyJSON(user))
-                        .omit("userCredentials")
-                        .imerge({ [props.field]: orgUnitsForUser.map(ou => ({ id: ou.id })) })
+                        .imerge({
+                            // getOwnedPropertyJSON returns only the id for user.userCredentials,
+                            // and dhis2 >= 2.32 needs the full object.
+                            userCredentials: user.userCredentials,
+                            [props.field]: orgUnitsForUser.map(ou => ({ id: ou.id }))
+                        })
                         .value()
                 ),
             }),

